Userland/Services/SystemServer/main.cpp

```diff
@@ -383,10 +383,7 @@ static ErrorOr<void> populate_devtmpfs()
 
 static ErrorOr<void> prepare_synthetic_filesystems()
 {
-    // FIXME: Don't hardcode the fs type as the ext2 filesystem and once there's
-    // more than this filesystem implementation (which is suitable for usage on
-    // physical storage), find a way to detect it.
-    TRY(Core::System::mount(-1, "/"sv, "ext2"sv, MS_REMOUNT | MS_NODEV | MS_NOSUID | MS_RDONLY));
+    TRY(Core::System::remount("/"sv, MS_NODEV | MS_NOSUID | MS_RDONLY));
```

Comment:
If the problem this PR is solving is this FIXME here, then can't we just make it so that when you pass `MS_REMOUNT`, we ignore the parameters that don't matter?

I'm not sure I agree with the rationale to split this up into distinct syscalls. It's more work for the `mount` program or any other userspace app that wants to interact with filesystem mounts. And it deviates from what all the other Unixes do for this operation that's been there since... [AT&T Unix Version 1](https://man.netbsd.org/mount.2#HISTORY)

Reply:
As far as I consider this, the `mount` syscall is just a hell of a mess that we should split up, especially when considering #16891 and #16299 as future work.

Linux had the `mount` syscall for a long time, but it now has multiple syscalls to handle the mount procedure in more "atomically"-built-sequence fashion. This is ideal in my opinion, because it basically means that handling the mount procedure becomes not just easier on the kernel side, but also means that debugging and figuring "why my new shiny mount at `/whatever` failed but I got this `errno` which doesn't say much" is not a problem anymore :)

Reply:
https://lwn.net/Articles/759499/
These are the new `mount`-related syscalls that linux has. I'd like to push forward to have something similar to this.